### PR TITLE
Adds additional cases for remove_redundant_buf optimization

### PR DESCRIPTION
**Pull request by sjalander**

### CODE REVIEW

File: jlm/hls/backend/rvsdg2rhls/remove-redundant-buf.cpp [hunk 1]
```diff
@@ -20,6 +20,11 @@ eliminate_buf(jlm::rvsdg::output * o)
     {
       return eliminate_buf(node->input(1)->origin());
     }
+    else if (dynamic_cast<const jlm::hls::fork_op *>(&node->GetOperation()))
```
Comment (phate) [inline jlm/hls/backend/rvsdg2rhls/remove-redundant-buf.cpp:23]: `is<>()`

File: jlm/hls/backend/rvsdg2rhls/remove-redundant-buf.cpp [hunk 2]
```diff
@@ -28,7 +33,16 @@ eliminate_buf(jlm::rvsdg::output * o)
     {
       return true;
     }
+    else if (dynamic_cast<const jlm::hls::load_op *>(&node->GetOperation()))
```
Comment (phate) [inline jlm/hls/backend/rvsdg2rhls/remove-redundant-buf.cpp:36]: `is<>()`

File: jlm/hls/backend/rvsdg2rhls/remove-redundant-buf.cpp [hunk 3]
```diff
@@ -28,7 +33,16 @@ eliminate_buf(jlm::rvsdg::output * o)
...
+    {
+      return true;
+    }
+    else if (dynamic_cast<const jlm::hls::store_op *>(&node->GetOperation()))
```
Comment (phate) [inline jlm/hls/backend/rvsdg2rhls/remove-redundant-buf.cpp:40]: `is<>()`

**Review by phate — CHANGES_REQUESTED** [review]
Again, could we get some unit tests on this?

File: jlm/hls/backend/rvsdg2rhls/remove-redundant-buf.cpp (hunk 2 above)
Comment (sjalander) [reply jlm/hls/backend/rvsdg2rhls/remove-redundant-buf.cpp]: Changed

File: jlm/hls/backend/rvsdg2rhls/remove-redundant-buf.cpp (hunk 3 above)
Comment (sjalander) [reply jlm/hls/backend/rvsdg2rhls/remove-redundant-buf.cpp]: Changed

**Review by phate — APPROVED** [review]
The unit tests were merged in another PR, right?

**Comment by sjalander** [comment]
A unit test has not been created for this PR. The reason is that when I started looking into the functionality of this optimization, I'm not able to determine its purpose. The name seems to be somewhat misleading. From what I can tell, it replaces pass through buffers with non-pass through buffers in some specific cases. However, it doesn't seem to have an effect on the hls_test_suite as when I comment out the optimization, then I get no change in the number of cycles.

@davidmetz Do you have a use case where this optimization makes a difference?

**Pull request merged**